helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/PropertyStoreAccessor.java

```diff
@@ -134,4 +135,28 @@ public Response putPropertyByPath(@PathParam("clusterId") String clusterId,
       return serverError(e);
     }
   }
+
+  /**
+   * Recursively deletes the PropertyStore path. If the node does not exist, it returns OK().
+   * @param clusterId
+   * @param path
+   * @return
+   */
+  @DELETE
+  @Path("{path: .+}")
+  public Response deletePropertyByPath(@PathParam("clusterId") String clusterId,
+      @PathParam("path") String path) {
+    path = "/" + path;
+    if (!ZkValidationUtil.isPathValid(path)) {
+      LOG.info("The propertyStore path {} is invalid for cluster {}", path, clusterId);
+      return badRequest(
+          "Invalid path string. Valid path strings use slash as the directory separator and names the location of ZNode");
+    }
+    final String recordPath = PropertyPathBuilder.propertyStore(clusterId) + path;
+    ZkBaseDataAccessor<byte[]> propertyStoreDataAccessor = getByteArrayDataAccessor();
+    if (!propertyStoreDataAccessor.remove(recordPath, AccessOption.PERSISTENT)) {
+      return serverError("Failed to delete PropertyStore record in path: " + path);
+    }
+    return OK();
```

Comment:
Nit, if I were a user, I would expect to see a clear response indicating the request is successful:
```
{
  "success": true
}
```
I think this is included in helix rest improvement plan.

Reply:
Let's discuss that as part of the rest improvement plan. We need to discuss what would make the most sense. 

This is done to keep it consistent with other DELETE endpoints.